packages/@expo/metro-config/proxies/NativeModules.js

```diff
@@ -0,0 +1,12 @@
+const NativeModules = require('react-native/Libraries/BatchedBridge/NativeModules');
+
+let proxy = NativeModules;
+
+try {
+  const { createProxyForNativeModules } = require('expo/build/proxies/NativeModules');
+  proxy = createProxyForNativeModules(NativeModules);
+} catch {
+  // do nothing, this is expected if no supported version of the expo package is installed
```

Comment:
This seems like not a great idea, we probably want to make use of all error messages and not introduce random error hiding.